src/Admin/Translation/Controller/MachineTranslationAdminController.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace App\Admin\Translation\Controller;
+
+use App\Commands\Helpers\CommandHelper;
+use Sonata\AdminBundle\Controller\CRUDController;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\KernelInterface;
+
+abstract class MachineTranslationAdminController extends CRUDController
+{
+  public const TYPE_PROJECT = 'TYPE_PROJECT';
+  public const TYPE_COMMENT = 'TYPE_COMMENT';
+
+  private string $type;
+
+  public function __construct(string $type)
```

Comment:
I couldn't figure out how to inject the type in `services.yaml` and then pass that to the constructor of the admin class, so I created 2 subclasses with different constructors.

Reply:
I guess using services or classes is just a matter of taste. However, it should have been possible by defining 2 named services using the same class but providing different arguments.
Something like this:

```
  projectTL:
    class: App\Translation\MachineTranslationListener
    arguments:
      - "type A"
      
  commenTL:
    class: App\Translation\MachineTranslationListener
    arguments:
      - "type B"    
```

And then reusing those services with something like
```
    arguments:
      - ~
      - @commentTL
      - \App\Admin\Translation\Controller\CommentMachineTranslationAdminController
```

However, I haven't used something like this in a long time. So a look into the documentation would still be necessary I guess :)

Reply:
I tried defining named services like that. But I think the constructor of `AbstractAdmin` takes a string of class name as the last argument. So I can't get that working by passing the service name as the class name.